lib/components/atoms/Pagination/Pagination.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+/**
+ *
+ * Pagination
+ *
+ */
+import React from 'react';
+import styled from 'styled-components';
+
+import styles from './Pagination.style';
+import type { Props } from './types';
+import Button from '../Button';
+
+const Pagination = ({
+  postsPerPage,
+  totalPosts,
+  paginate,
+  className,
+  firstLabel,
```

Comment:
Looks like you are not making it as HOC, Fine than it this case your component should be much more scalable and follow declarative style.